flink/src/main/java/org/apache/iceberg/flink/source/DataIterator.java

```diff
@@ -79,6 +83,27 @@ InputFile getInputFile(String location) {
     return inputFiles.get(location);
   }
 
+  public void seek(CheckpointedPosition checkpointedPosition)  {
```

Comment:
That makes sense to me.

I have a question about the `Map<String, InputFile> inputFiles`. Right now, it is constructed per `CombinedScanTask`. Would it be ok to do it at each individual `FileScanTask`? I tried the change and delete tests work fine. But I am not sure if I could miss anything since I am not familiar on read merge on deleted rows.

Reply:
@openinx can you take a look at my question in the comment above?

Reply:
The `inputFiles` in DataIterator is a in-memory cache to get the given decrypted InputFile path by decrypted location.  We maintain those <location,  decryptedInputFile> into map because we are trying to get all the decrypted infos once ( Some `EncryptionManager` implementation will use this feature to request them in a batch RPC call).  It don't have relationship to row-level delete in format v2,  we could fetch the <location, decryptedInputFile> one by one but that may produce many RPC to a key server.

Reply:
I made the change in the child PR #2305 . In particular, this is the commit: https://github.com/apache/iceberg/pull/2305/commits/cec66f9148ffac666117125ca2af2139ccde839c

I haven't merged it to this Uber PR yet, since I am hoping to get it reviewed first in the child PR #2305.